### PR TITLE
Update samples to SDK version 2.4.0

### PR DESCRIPTION
**Pull request by SatjaSivcev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,6 +147,18 @@ Zivid Samples are distributed under the [BSD license](source/LICENSE).
...
 [color_balance-url]: source/applications/advanced/color_balance.py
-[mask_point_cloud-url]: source/applications/advanced/mask_point_cloud.py
+
+[open-3d-url]: http://www.open3d.org/
+[numpy-url]: https://numpy.org/
+[matplotlib-url]: https://matplotlib.org/
+[openCV-url]: https://opencv.org/
+[scipy-url]: https://www.scipy.org/
+
+[knowladge-base-url]: https://support.zivid.com/
```
Comment (knzivid) [inline README.md:160]: typo

Comment (SatjaSivcev) [reply README.md]: fixed, sorry about that.

**Comment by SatjaSivcev** [comment]
All should work now @knzivid

**Review by chrisasc — DISMISSED** [review]
lgtm

**Review by chrisasc — COMMENTED** [review]
I have no idea if this is the right way of passing the sample code through CI or not, but the rest still looks good for me! Maybe Karthik can give a green light in the changes in the last commit?

**Review by chrisasc — DISMISSED** [review]

**Comment by knzivid** [comment]
LGTM. Does the pylint ignore commit have a design discussion thread? @trym-b should green light it otherwise

**Comment by SatjaSivcev** [comment]
> LGTM. Does the pylint ignore commit have a design discussion thread? @trym-b should green light it otherwise

No design discussion. Each code sample is standalone and similarities (duplication) are expected to happen. It is unclear why the CI/Pylint was not complaining before though.

**Comment by trym-b** [comment]
> No design discussion. Each code sample is standalone and similarities (duplication) are expected to happen. It is unclear why the CI/Pylint was not complaining before though. I will wait to hear your opinion @trym-b.

Seeing that master has not been built since 2 months ago a lot of things could have happened. I do not know why it specifically fails though.

Seems strange that you want to ignore the imports, perhaps the code analysis setup is missing some package?

**Comment by eskaur** [comment]
> > LGTM. Does the pylint ignore commit have a design discussion thread? @trym-b should green light it otherwise
> 
> No design discussion. Each code sample is standalone and similarities (duplication) are expected to happen. It is unclear why the CI/Pylint was not complaining before though. I will wait to hear your opinion @trym-b.

It probably suddenly started complaining for these reasons:
- `zivid-python-samples/continuous-integration/requirements.txt` does not specify versions for the linting tools, which means the CI will just install the latest versions.
- Due to inactivity the workflows have not been run in a couple of months. The tools have probably been changed in the meantime.

**Review by eskaur — DISMISSED** [review]

**Comment by SatjaSivcev** [comment]
@trym-b @eskaur I removed ignoring imports. Please review/approve again.

**Review by trym-b — APPROVED** [review]
Looks good to me.

**Pull request merged**